app/controllers/general.php

```diff
@@ -357,6 +358,15 @@ function router(App $utopia, Database $dbForConsole, callable $getProjectDB, Swo
     return false;
 }
 
+App::init()
+    ->groups(['database', 'function', 'storage', 'messaging'])
+    ->inject('project')
+    ->action(function (Document $project) {
+        if ($project->getId() === 'console') {
+            throw new Exception(Exception::PROJECT_RESERVED_PROJECT, 'Please check X-Appwrite-Project header');
```

Comment:
This error is not good enough. It assumes the consumer context is the API but ignores SDKs or console where the consumer has no idea what this error is.

Reply:
@gewenyu99 Any input for a message here? This is the case when the project is set as `console`, and you're calling an unsupported API like databases, functions etc.

Reply:
Questions here: 

1. Can I name my project ID `console`? We shouldn't allow this during creation.
2. Could this ever display on the console?

Either way:
[What went wrong][why we think it went wrong][steps to debug/fix]
We can try:
`This endpoint is not available for the `console` project. The Appwrite Console is a reserved project ID and cannot be used with the Appwrite SDKs and APIs. Please check if your project ID is correct.`

Reply:
@gewenyu99 No we can not you will get an error this project name is resolved.

I think this message is misleading since in 99% cases when  X-Appwrite-Project Header is not assigned, and 1%  the project 'console' is assigned, so we have a fallback to console default project.

Reply:
@gewenyu99 
1. You can not name a project ID console, it's disallowed during creation.
2. This happens in 99% when a user does not assign the X-Appwrite-Project header , We are the ones who fallback to console project

Reply:
@fogelito We should throw a separate error for when project ID is not assigned. Is that possible?

Reply:
I guess we can do something like:
https://github.com/appwrite/appwrite/pull/7341/files#diff-ff12d23a7b6584b45f7e669b24ae1637106903cbac9390cad416e2c21dfda3f4R367

@abnegate what do you think?

Reply:
@fogelito 
Let's work on the error message:
IF empty:
`No Appwrite project was specified. Please specify your Appwrite Project ID in the `x-appwrite-project` header or when you initialize your Appwrite SDK Client.

Reply:
If console:
`This endpoint is not available for the console project. The Appwrite Console is a reserved project ID and cannot be used with the Appwrite SDKs and APIs. Please check if your project ID is correct.`

Reply:
1. The console project is something internal, a normal dev using Appwrite has no idea what it is and its very confusing. 
2. We shouldn't mention `x-appwrite-project` just like that, this header is only visible to REST users, which are minority if we do mention the context should be clear and the first thing mention should be setting the project ID. We can also skip the header name altogether.

Reply:
1. @eldadfux We have two errors now. IF they get an error that mentions console, they have it set as their project. IMO this is fine.

2. We can do: No Appwrite project was specified. Please specify your Appwrite Project ID on your Appwrite SDK Client.